koku/masu/external/downloader/ibm/ibm_report_downloader.py

```diff
@@ -0,0 +1,242 @@
+#
+# Copyright 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""IBM Report Downloader."""
+import logging
+import os
+
+from django.conf import settings
+from ibm_cloud_sdk_core import BaseService
+from ibm_cloud_sdk_core.api_exception import ApiException
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_platform_services.common import get_sdk_headers
+from rest_framework.exceptions import ValidationError
+
+from api.common import log_json
+from api.provider.models import Provider
+from api.utils import DateHelper
+from masu.config import Config
+from masu.external import UNCOMPRESSED
+from masu.external.downloader.downloader_interface import DownloaderInterface
+from masu.external.downloader.report_downloader_base import ReportDownloaderBase
+from masu.util.aws.common import copy_local_report_file_to_s3_bucket
+from masu.util.common import get_path_prefix
+from masu.util.ibm import common as util
+from providers.ibm.provider import IBMProvider
+
+
+DATA_DIR = Config.TMP_DIR
+LOG = logging.getLogger(__name__)
+REPORTS_DIR = Config.INSIGHTS_LOCAL_REPORT_DIR
+
+
+class NoMoreReportPages(Exception):
+    pass
+
+
+def page_downloader_factory(credentials, data_source, month):
+    service_version = "v1"
+    service_name = "resource-instance-usage-reports"
+    try:
+        authenticator = IAMAuthenticator(credentials.get("iam_token", ""))
+        service = BaseService(service_url="https://enterprise.cloud.ibm.com", authenticator=authenticator)
```

Comment:
We don't have to do this now, but we might want to consider making this be an environment variable.